rewrite-core/src/main/java/org/openrewrite/config/Environment.java

```diff
@@ -28,18 +29,26 @@
 import java.time.Duration;
 import java.util.*;
 
+import static java.util.Collections.emptyList;
 import static java.util.stream.Collectors.toList;
 
 public class Environment {
     private final Collection<? extends ResourceLoader> resourceLoaders;
+    private final Collection<? extends ResourceLoader> dependencyResourceLoaders;
 
     public Collection<Recipe> listRecipes() {
+        List<Recipe> dependencyRecipes = dependencyResourceLoaders.stream()
+                .flatMap(r -> r.listRecipes().stream())
+                .collect(toList());
         List<Recipe> recipes = resourceLoaders.stream()
                 .flatMap(r -> r.listRecipes().stream())
                 .collect(toList());
         for (Recipe recipe : recipes) {
             if (recipe instanceof DeclarativeRecipe) {
-                ((DeclarativeRecipe) recipe).initialize(recipes);
+                List<Recipe> availableRecipes = new ArrayList<>();
```

Comment:
I think we can move the allocation of available recipes outside the for loop and do this only once?

Reply:
yeah good call, i'll add that in a follow-up commit after the ci finishes